File: leetCode/21. 合并两个有序链表/Main.py
class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head = ListNode(None)
        cur = head
        while l1 or l2:
            if not l1:
                cur.next = l2
                break
            if not l2:
                cur.next = l1
                break
            if l1.val < l2.val:
                cur.next = l1
                l1 = l1.next
            else:
                cur.next = l2
                l2 =l2.next
            cur = cur.next
        return head.next

File: leetCode/21. 合并两个有序链表/Main.java
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if(l1 == null)return l2;
        if(l2 == null)return l1;
        ListNode left;
        ListNode head;
        if(l1.val <= l2.val){
            head = l1;
            left = l1;
            l1 = l1.next;
        }else{
            head = l2;
            left = l2;
            l2 = l2.next;
            head.next = l1;
        }
        while(l1 != null || l2 != null){
            if(l1 == null){
                left.next = l2;
                break;
            }else if(l2 == null){
                break;
            }
            if(l1.val <= l2.val){
                left = l1;
                l1 = l1.next;
            }else{
                left.next = l2;
                l2 = l2.next;
                left.next.next = l1;        
                left = left.next;
            }
        }
        return head;
    }
}
